test/middleware_stack_test.rb

```diff
@@ -35,8 +35,9 @@ def test_allows_rebuilding
   end
 
   def test_allows_extending
-    build_stack Apple
+    build_handlers_stack Apple
     @conn.use Orange
+    @conn.adapter :test, &test_adapter
     assert_handlers %w[Apple Orange]
```

Comment:
I did look to parameterize `build_stack` but the splat argument makes this awkward. If we were targeting Ruby >= 2.0 we could use keyword arguments like so:

```
def build_stack(*handlers, without_adapter: true)
```

But this fails on Ruby 1.9.3. I'll find another solution to reduce the duplication.